Tests/Common/Background Queue/QueueRunRequestTest.swift

```diff
@@ -119,3 +119,125 @@ class QueueRunRequestTest: UnitTest {
         XCTAssertEqual(storageMock.deleteCallsCount, 2)
     }
 }
+
+class QueueRunRequestIntegrationTest: IntegrationTest {
```

Comment:
Added integration tests to be more confident with the BQ refactor that I did. 

We only mock the queue runner so API calls are not actually performed.